src/Config.php

```diff
@@ -3750,4 +3750,111 @@ final public static function generateUuid($type)
         self::setConfigurationValues('core', [$type . '_uuid' => $uuid]);
         return $uuid;
     }
+
+    /**
+     * Get email and name to be used as "From" in our email notifications
+     *
+     * @param int  $entities_id
+     * @param bool $allow_response
+     *
+     * @return array [email => sender address, name => sender name]
+     */
+    public static function getAdminEmailSender(
+        int $entities_id,
+        bool $allow_response
+    ): array {
+        if ($allow_response) {
+            return self::getAdminEmail($entities_id);
+        } else {
+            return self::getNoReplyEmail();
+        }
```

Comment:
In previous logic, the "admin email" value was returned as a fallback when `admin_email_noreply` was not defined in `!$allow_response` mode.
I guess we should keep this fallback.
```suggestion
        if (!$allow_response) {
            $admin = self::getNoReplyEmail();
            if ($admin['email'] !== null) {
                return $admin;
            }
        }
        return self::getAdminEmail($entities_id);
```

Reply:
Not sure of this. My previous review was proposing the opposite.

Reply:
Yes, that's why I changed the function in the first place.
Which behavior do you want to keep ?

Reply:
Actually, the current method does not make sense. If you need the admin email, you can directly call the `getAdminEmail()` method, and if you need the no-reply email, you can call the `getNoReplyEmail()` method.
See my comments in `NotificationTarget::getSender()` method.